### PR TITLE
LPS-171480 Removing CompanyThreadLocal value setting as we are initializing the portal and we need to be in SYSTEM

### PR DESCRIPTION
**Pull request by LuisOrtiz89**

### CODE REVIEW

**Comment by LuisOrtiz89** [comment]
Hi @shuyangzhou, this tickets solves the issue discovered after [LPS-159081](https://issues.liferay.com/browse/LPS-159081).

The thing is that now, when auto upgrade on startup is enabled, the loading of the components of the portal is done in just one thread. The MainServlet._initCompanies method ends by setting the CompanyThreadLocal to the default companyId, which we think is wrong, as we are not in any company, we are just starting up the portal.

As now there is only one thread in that case, when the MessageListeners activate, they have the CompanyThreadLocal set to the default company ID instead to SYSTEM, so they are registered with that ID. We need they to register with Company ID SYSTEM, so the [DBPartitionMessageBusInterceptor](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal/portal-db-partition/src/main/java/com/liferay/portal/db/partition/internal/messaging/DBPartitionMessageBusInterceptor.java#L53) can generate as many messages as companies are registered.

**Comment by catarineaguiar** [comment]
no regressions found.

File: modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+public class SchedulerResponseTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testStartupScheduleJobCompanyId() throws SchedulerException {
+		SchedulerResponse schedulerResponse =
```
Comment (tinatian) [inline modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java:46]: @LuisOrtiz89 Scheduled jobs itself has nothing related to company, it is that when a job is triggered, it includes the company context. You can not test like this as next time the company id may be changed, it depends on the runtime context

Comment (LuisOrtiz89) [reply modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java:46]: Hi @tinatian, for what we saw, when some MessageListener are activated they create a trigger and register an entry in the scheduler. You can see an example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/messaging/CheckCommerceDiscountMessageListener.java).

In those cases, when the scheduled job is triggered the message comes populated with a company ID that corresponds with the company ID set in the CompanyThreadLocal at the moment of the component activation. 

We need that this company ID is SYSTEM, as it indicates that the message is not directed to any company in particular, so the [DBPartitionMessageBusInterceptor](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal/portal-db-partition/src/main/java/com/liferay/portal/db/partition/internal/messaging/DBPartitionMessageBusInterceptor.java#L53) can generate as many messages as company IDs are registered.

Comment (LuisOrtiz89) [reply modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java:46]: Hi @tinatian, is everything clear? Did you were able to take another look at it?

Comment (LuisOrtiz89) [reply modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java]: Hi @tinatian, as discussed, I removed the test and changed the commit and PR message.
You can see the previously QA review in this PR: https://github.com/liferay-uniform/liferay-portal/pull/869

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/128186

**Pull request closed without merging**